src/layout/col-div.tsx

```diff
@@ -0,0 +1,95 @@
+import React from "react";
+import { ColDivProps } from "./types";
+import { StyledDiv } from "./col-div.style";
+import { BreakpointValues } from "../theme/breakpoint/theme-helper";
+import { ThemeSpec } from "../theme/types";
+import { useTheme } from "styled-components";
+
+const Component = (
+    props: ColDivProps,
+    ref: React.Ref<HTMLDivElement>
+): JSX.Element => {
+    const theme = useTheme() as ThemeSpec;
+
+    const {
+        xxlCols,
+        xlCols,
+        lgCols,
+        mdCols,
+        smCols,
+        xsCols,
+        xxsCols,
+        ...otherProps
+    } = props;
+
+    const getColSpan = (
+        cols: number | [number, number] | undefined,
+        maxCols: number,
+        label: string
+    ) => {
+        if (!cols) return { start: undefined, span: undefined };
+
+        // during development process to give wwarning
+        if (process.env.NODE_ENV === "development") {
+            if (typeof cols === "number" && cols > maxCols) {
+                console.warn(
+                    `Warning: ${label}Cols exceeds maximum (${maxCols}): ${cols}`
+                );
+            } else if (
+                Array.isArray(cols) &&
+                (cols[0] > maxCols + 1 || cols[1] > maxCols + 1)
+            ) {
+                console.warn(
+                    `Warning: ${label}Cols array exceeds maximum (${maxCols}): [${cols[0]}, ${cols[1]}]`
+                );
+            }
+        }
+
+        if (Array.isArray(cols)) {
+            const [start, end] = cols;
+            const span = Math.min(end - start, maxCols);
+            return { start, span };
+        }
+
+        return { start: undefined, span: Math.min(cols, maxCols) };
+    };
+
+    const getStyleProps = () => {
+        const xxlColumnCount = BreakpointValues["xxl-column"]({ theme });
+        const xlColumnCount = BreakpointValues["xl-column"]({ theme });
+        const lgColumnCount = BreakpointValues["lg-column"]({ theme });
+        const mdColumnCount = BreakpointValues["md-column"]({ theme });
+        const smColumnCount = BreakpointValues["sm-column"]({ theme });
+        const xsColumnCount = BreakpointValues["xs-column"]({ theme });
+        const xxsColumnCount = BreakpointValues["xxs-column"]({ theme });
+
+        const xxlStartSpan = getColSpan(xxlCols, xxlColumnCount, "xxl");
+        const xlStartSpan = getColSpan(xlCols, xlColumnCount, "xl");
+        const lgStartSpan = getColSpan(lgCols, lgColumnCount, "lg");
+        const mdStartSpan = getColSpan(mdCols, mdColumnCount, "md");
+        const smStartSpan = getColSpan(smCols, smColumnCount, "sm");
+        const xsStartSpan = getColSpan(xsCols, xsColumnCount, "xs");
+        const xxsStartSpan = getColSpan(xxsCols, xxsColumnCount, "xxs");
```

Comment:
the original implementation also supports fallback behaviour e.g. desktop will automatically inherit tablet if not specified

to restore this functionality